modules/virt-about-pci-passthrough.adoc

```diff
@@ -3,8 +3,35 @@
 // * virt/virtual_machines/advanced_vm_management/virt-configuring-pci-passthrough.adoc
 
 [id="virt-about_pci-passthrough_{context}"]
-= About preparing nodes and the cluster for PCI passthrough
+= About preparing a host device for PCI passthrough
 
-The Peripheral Component Interconnect (PCI) passthrough feature enables you to access and manage hardware devices from a virtual machine. When PCI passthrough is configured, the PCI devices function as if they were physically attached to the virtual machine.
+To prepare a host device for PCI passthrough using the CLI, create a `MachineConfig` object and add kernel arguments to enable the Input-Output Memory Management Unit (IOMMU). Bind the PCI device to the Virtual Function I/O (VFIO) driver and then expose it in the cluster by editing the `permittedHostDevices` field of the `HyperConverged` custom resource (CR). The `permittedHostDevices` list is empty when you first install the HyperConverged Cluster Operator.
 
-To configure a hardware device as a host device for PCI passthrough, create a `MachineConfig` object file and add kernel arguments to enable Input-Output Memory Management Unit (IOMMU). Bind the PCI device to the Virtual Function I/O (VFIO) driver by creating a `MachineConfig` object file. Then, add the PCI device to the HyperConverged Operator.
+The following example snippet shows two PCI host devices that are named `nvidia.com/GV100GL_Tesla_V100` and `nvidia.com/TU104GL_Tesla_T4` added to the list of permitted host devices in the `HyperConverged` CR.
+
+[source,yaml]
+----
+apiVersion: hco.kubevirt.io/v1beta1
+kind: HyperConverged
+metadata:
+  name: kubevirt-hyperconverged
+spec:
+  permittedHostDevices: <1>
+    pciHostDevices: <2>
+    - pciDeviceSelector: "10DE:1DB6" <3>
+      resourceName: "nvidia.com/GV100GL_Tesla_V100" <4>
+    - pciDeviceSelector: "10DE:1EB8"
+      resourceName: "nvidia.com/TU104GL_Tesla_T4"
+...
```

Comment:
Why do we show these changes in the about section? Isn't this redundant to the snippet in the "Exposing PCI host devices in the cluster using the CLI" section?

Reply:
@fabiand The reason I mentioned the two Nvidia devices upfront is because, from what I understand, the admin does not need to go through the procedures to add kernel arguments and bind the devices to the VFIO driver. They can directly edit the HCO CR and expose these devices in the cluster. For any other PCI host device, the admin has to go through the whole process. I wanted to point out that difference.

Also, the example snippet in the "Exposing PCI host devices..." section shows a user-added device with the `externalResourceProvider `field set to `true` which is not used for the Nvidia devices. 

Does that make sense?

Reply:
> @fabiand The reason I mentioned the two Nvidia devices upfront is because, from what I understand, the admin does not need to go through the procedures to add kernel arguments and bind the devices to the VFIO driver. They can directly edit the HCO CR and expose these devices in the cluster. For any other PCI host device, the admin has to go through the whole process. I wanted to point out that difference.
> 
@sjhala-ccs Unfortunately, that's incorrect. Any device PCI device that is intended to be assignable to Virtual Machines needs to be bound to the VFIO driver. This includes these two devices.

> Also, the example snippet in the "Exposing PCI host devices..." section shows a user-added device with the `externalResourceProvider `field set to `true` which is not used for the Nvidia devices.

The `externalResourceProvider ` field indicates whether this resource is being provided by an external device plugin. If it's set to `true`, KubeVirt/CNV will only permit the usage of this device in the cluster but will leave the allocation and monitoring to an external device plugin.



> 
> Does that make sense?